pkg/ingester/ingester.go

```diff
@@ -519,8 +519,13 @@ func (i *Ingester) updateLoop(ctx context.Context) error {
 	usageStatsUpdateTicker := time.NewTicker(usageStatsUpdateInterval)
 	defer usageStatsUpdateTicker.Stop()
 
+	inflightRequestsTicker := time.NewTicker(250 * time.Millisecond)
```

Comment:
I think you will need a dedicated goroutine for this ticker, because we don't want other logic to slow it down.